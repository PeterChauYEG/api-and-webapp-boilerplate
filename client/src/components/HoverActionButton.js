import React from 'react'

// import css
import '../CSS/HoverActionButton.css'

const HoverActionButton = (props) => {
  const {
    action,
    backgroundColor,
    text,
  } = props

  return (
    <div className={`HoverActionButton ${backgroundColor}`}
      onClick={action} >
      <p>{text}</p>
    </div>
  );
}

HoverActionButton.propTypes = {
  action: React.PropTypes.func.isRequired,
  backgroundColor: React.PropTypes.string.isRequired,
  text: React.PropTypes.string.isRequired,
}

export default HoverActionButton;
