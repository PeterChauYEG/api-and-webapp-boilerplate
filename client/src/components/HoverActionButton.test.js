import React from 'react'

// import test dependencies 
import renderer from 'react-test-renderer'

// import component
import HoverActionButton from './HoverActionButton'

// render test
test('renders without crashing', () => {
  
  // declare prop
  const action = () => {}
  const backgroundColor = ""
  const text = ""
  
  // render component
  const component = renderer.create(
    <HoverActionButton 
      action={action} 
      backgroundColor={backgroundColor} 
      text={text}
      />
  )

  // render DOM tree
  const tree = component.toJSON()
  
  // assertion
  expect(tree).toMatchSnapshot()
  
});

// onClick test
test('onClick callback is called', () => {
  
  // declare prop
  const action = () => {}
  const backgroundColor = ""
  const text = ""
  
  // render component
  const component = renderer.create(
    <HoverActionButton 
      action={action} 
      backgroundColor={backgroundColor} 
      text={text}
      />
  )

  // render DOM tree
  let tree = component.toJSON()
  
  // manually trigger the callback
  tree.props.onClick()
  
  // re-rendering
  tree = component.toJSON()
  
  // assertion
  expect(tree).toMatchSnapshot()
  
});