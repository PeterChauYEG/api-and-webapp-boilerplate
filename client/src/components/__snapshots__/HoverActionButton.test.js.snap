exports[`test onClick callback is called 1`] = `
<div
  className="HoverActionButton "
  onClick={[Function]}>
  <p>
    
  </p>
</div>
`;

exports[`test renders without crashing 1`] = `
<div
  className="HoverActionButton "
  onClick={[Function]}>
  <p>
    
  </p>
</div>
`;
